packages/wdio-types/src/Services.ts

```diff
@@ -234,17 +234,19 @@ export interface HookFunctions {
      * beforeEach in Mocha). `stepData` and `world` are Cucumber framework specific properties.
      * @param test      details to current running test (represents step in Cucumber)
      * @param context   context to current running test (represents World object in Cucumber)
-     */
-    beforeHook?(test: any, context: any): unknown | Promise<unknown>
+     * @param hookName  name of the hook
+    */
+    beforeHook?(test: any, context: any, hookName: string): unknown | Promise<unknown>
 
     /**
      * Hook that gets executed _after_ a hook within the suite ends (e.g. runs after calling
      * afterEach in Mocha). `stepData` and `world` are Cucumber framework specific.
      * @param test      details to current running test (represents step in Cucumber)
      * @param context   context to current running test (represents World object in Cucumber)
+     * @param hookName  name of the hook
      * @param result    test result
-     */
-    afterHook?(test: Test, context: any, result: TestResult): unknown | Promise<unknown>
+    */
+    afterHook?(test: Test, context: any, hookName: string, result: TestResult): unknown | Promise<unknown>
```

Comment:
This could be a potentially breaking change for user that access the `result` object, how about we also add it as last argument?

Reply:
i'm trying to do so, but i'm getting an compiling issue with this part:
<img width="859" alt="image" src="https://github.com/webdriverio/webdriverio/assets/48208030/a20ddadd-96d3-4ce7-b937-fb7882f3d588">

any idea on what to do here?

Reply:
```
        const emitHookEvent = (
            fnName: string,
            eventType: string
        ) => (
            _test: never,
            _context: never,
            { error }: { error?: jasmine.FailedExpectation } = {},
            _hookName: never
```

Reply:
Just leave out the parameter

Reply:
Small reminder to update the docs/templates since the argument is now last